tests/models/test_layout.py

```diff
@@ -19,25 +19,198 @@
+    self.gpg_keyid2 = "8288ef560ed3795f9df2c0db56193089b285da58"
+    self.test_dir = os.path.realpath(tempfile.mkdtemp())
+
+    # Create directory to run the tests without having everything blow up
+    self.working_dir = os.getcwd()
+
+    # Copy GPG keyring to temp test dir
+    tests_dir = os.path.join(os.path.dirname(os.path.realpath(__file__)), "..")
+
+    self.gnupg_home = os.path.join(self.test_dir, "rsa")
+    shutil.copytree(
+        os.path.join(tests_dir, "gpg_keyrings", "rsa"),
+        self.gnupg_home)
+
+    # Copy keys to temp test dir
+    key_names = ["bob", "bob.pub", "carl.pub"]
+    for name in key_names:
+      shutil.copy(
+        os.path.join(tests_dir, "demo_files", name),
+        os.path.join(self.test_dir, name))
+
+    self.key_path = os.path.join(self.test_dir, "bob")
+    self.pubkey_path1 = os.path.join(self.test_dir, "bob.pub")
+    self.pubkey_path2 = os.path.join(self.test_dir, "carl.pub")
+
+    os.chdir(self.test_dir)
+
+
+
+  @classmethod
+  def tearDownClass(self):
+    """Change back to initial working dir and remove temp test directory. """
+    os.chdir(self.working_dir)
+    shutil.rmtree(self.test_dir)
+
+
+  def test_set_relative_expiration(self):
+    """Test adding expiration date relative from today. """
+    layout = Layout()
+    with self.assertRaises(securesystemslib.exceptions.FormatError):
+      layout.set_relative_expiration(days=None, months=0, years=0)
+
+    with self.assertRaises(securesystemslib.exceptions.FormatError):
+      layout.set_relative_expiration(days=0, months="", years=0)
+
+    with self.assertRaises(securesystemslib.exceptions.FormatError):
+      layout.set_relative_expiration(days=0, months=0, years=[])
+
+    layout.set_relative_expiration(days=1)
+    layout._validate_expires()
+    layout.set_relative_expiration(months=2)
+    layout._validate_expires()
+    layout.set_relative_expiration(years=3)
+    layout._validate_expires()
+    layout.set_relative_expiration(days=3, months=2, years=1)
+    layout._validate_expires()
+
+
+  def test_get_step_name_list(self):
+    """Test getting list of step names. """
+    names = ["a", "b", "c"]
+    layout = Layout(steps=[Step(name=name) for name in names])
+    self.assertListEqual(layout.get_step_name_list(), names)
+
+
+  def test_get_step_by_name(self):
+    """Test getting step by name. """
+    names = ["a", "b", "c"]
+    layout = Layout(steps=[Step(name=name) for name in names])
+    self.assertEqual(layout.get_step_by_name("b").name, "b")
+
+    with self.assertRaises(securesystemslib.exceptions.FormatError):
+      layout.get_step_by_name(None)
+
+
+  def test_remove_step_by_name(self):
+    """Test removing step by name. """
+    names = ["a", "b", "c"]
+    layout = Layout(steps=[Step(name=name) for name in names])
+    layout.remove_step_by_name("b")
+    self.assertEqual(len(layout.steps), 2)
+    self.assertTrue("b" not in layout.get_step_name_list())
```

Comment:
Can you try a double-remove?

Reply:
Sure. What's expected behavior? Currently it would do nothing.